platform/packages/versioning/src/release.rs

```diff
@@ -1,279 +1,268 @@
 use serde::{Deserialize, Serialize};
 
 use sdk::{
-    cosmwasm_std::StdError,
+    cosmwasm_std::{StdError, Storage},
+    cw_storage_plus::Item,
     schemars::{self, JsonSchema},
 };
 
-use super::{Version, VersionSegment};
+use crate::SemVer;
+
+use super::Version;
 
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, JsonSchema)]
 #[repr(transparent)]
 #[serde(transparent)]
-pub struct Release(String);
+pub struct Id(String);
+
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "schema", derive(JsonSchema))]
+#[cfg_attr(test, derive(Debug))]
+pub struct Release {
+    id: Id,
+    code: Version,
+}
 
-impl Release {
-    const RELEASE_LABEL: &'static str = env!(
-        "RELEASE_VERSION",
-        "No release label provided as an environment variable! Please set \
-        \"RELEASE_VERSION\" environment variable!",
+impl Id {
+    const ID: &'static str = env!(
+        "SOFTWARE_RELEASE_ID",
+        "No software release identifier provided as an environment variable! Please set \
+        \"SOFTWARE_RELEASE_ID\" environment variable!",
     );
 
-    const DEV_RELEASE: &'static str = "dev-release";
+    //TODO delete once deliver a version with contracts that take the current release as input
+    const PREV: &'static str = "v0.7.6";
+
+    const DEV: &'static str = "dev-release";
+
+    const VOID: &'static str = "void-release";
+
+    fn this() -> Self {
+        Self(Self::ID.into())
+    }
+
+    fn prev() -> Self {
+        Self(Self::PREV.into())
+    }
 
-    const VOID_RELEASE: &'static str = "void-release";
+    fn dev() -> Self {
+        Self(Self::DEV.into())
+    }
 
+    fn void() -> Self {
+        Self(Self::VOID.into())
+    }
+}
+
+impl Release {
     pub fn void() -> Self {
```

Comment:
This method should be marked as `const` as it is a pure and total/non-divergent function, not even relying on heap operations.
```suggestion
    pub const fn void() -> Self {
```

Reply:
I see now. I must have missed that detail on when I wrote it.
I would suggest refactoring it to use `Cow<'static, str>` as it would act like a `String` during deserialization, yet not come with the additional overhead of memory allocations. Doing so will also solve this problem here as we know this data statically.

Reply:
done